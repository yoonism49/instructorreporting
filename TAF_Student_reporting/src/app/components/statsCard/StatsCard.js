// @flow weak

import React      from 'react';
import PropTypes  from 'prop-types';

const StatsCard = ({
  statValue,
  statLabel,
  icon,
  backColor
}) => (
  <div className="sm-st clearfix">
    <span className={ `sm-st-icon st-${backColor}` }>
      { icon }
    </span>
    <div className="sm-st-info">
      <span>
        { statValue }
      </span>
      <div>
        { statLabel }
      </div>
    </div>
  </div>
);

StatsCard.propTypes = {
  statValue:  PropTypes.string,
  statLabel:  PropTypes.string,
  icon:       PropTypes.node,
  backColor:  PropTypes.oneOf([
    'red',
    'blue',
    'violet',
    'green'
  ])
};

StatsCard.defaultProps = {
  statValue:  '0',
  statLabel:  'unknown',
  icon:       (<i className="fa fa-check-square-o"></i>),
  backColor:  'blue'
};

export default StatsCard;
