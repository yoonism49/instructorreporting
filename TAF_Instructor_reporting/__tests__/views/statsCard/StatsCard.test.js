
// @flow weak

import React            from 'react';
import StatsCard        from '../../../src/app/views/statsCard/statsCard';
import renderer         from 'react-test-renderer'; // needed both for snpashot testing but also to prevent errors from enzyme
import {
  shallow
}                       from 'enzyme';
import { 
  MemoryRouter
}                       from 'react-router';

// react-highlight uses findDOMNode: 
// -> jest will throw errors (no way to counter that...)
jest.mock('react-highlight');


describe('StatsCard view', () => {
  it('renders as expected', () => {
    const props = {
      actions: {
        enterStatsCard: () => {},
        leaveStatsCard: () => {}
      }
    };
    const component = renderer.create(
      <div>
        <MemoryRouter>
          <StatsCard {...props} />
        </MemoryRouter>
      </div>
    ).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('triggers enterStatsCard on mount', () => {
    const mockenterStat = jest.fn();
    const mockProps = {
      actions: {
        enterStatsCard: mockenterStat,
        leaveStatsCard: () => {}
      }
    };
    /* eslint-disable no-unused-vars */
    const wrapper = shallow(
      <StatsCard {...mockProps} />
    );
    expect(mockenterStat.mock.calls.length).toBe(1);
  });

  it('triggers leaveStatsCard on unMount', () => {
    const mockleaveStat = jest.fn();
    const mockProps = {
      actions: {
        enterStatsCard: () => {},
        leaveStatsCard: mockleaveStat
      }
    };
    const wrapper = shallow(
      <StatsCard {...mockProps} />
    );
    wrapper.unmount();
    expect(mockleaveStat.mock.calls.length).toBe(1);
  });
});
