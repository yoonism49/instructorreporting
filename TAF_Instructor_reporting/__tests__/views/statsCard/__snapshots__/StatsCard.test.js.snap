// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatsCard view renders as expected 1`] = `
<div>
  <section
    className="content view-enter"
  >
    <div
      className="row"
    >
      <div
        className="col-xs-12"
      >
        <section
          className="panel "
        >
          <header
            className="panel-heading"
          >
            Stats cards
          </header>
          <div
            className="panel-body undefined"
            style={
              Object {
                "backgroundColor": "#F4F5F6",
              }
            }
          >
            <div
              className="row"
            >
              <div
                className="col-md-3"
              >
                <div
                  className="sm-st clearfix"
                >
                  <span
                    className="sm-st-icon st-red"
                  >
                    <i
                      className="fa fa-check-square-o"
                    />
                  </span>
                  <div
                    className="sm-st-info"
                  >
                    <span>
                      3200
                    </span>
                    <div>
                      Total Tasks
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="col-md-3"
              >
                <div
                  className="sm-st clearfix"
                >
                  <span
                    className="sm-st-icon st-violet"
                  >
                    <i
                      className="fa fa-envelope-o"
                    />
                  </span>
                  <div
                    className="sm-st-info"
                  >
                    <span>
                      2200
                    </span>
                    <div>
                      Total Messages
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="col-md-3"
              >
                <div
                  className="sm-st clearfix"
                >
                  <span
                    className="sm-st-icon st-blue"
                  >
                    <i
                      className="fa fa-dollar"
                    />
                  </span>
                  <div
                    className="sm-st-info"
                  >
                    <span>
                      100,320
                    </span>
                    <div>
                      Total Profit
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="col-md-3"
              >
                <div
                  className="sm-st clearfix"
                >
                  <span
                    className="sm-st-icon st-green"
                  >
                    <i
                      className="fa fa-paperclip"
                    />
                  </span>
                  <div
                    className="sm-st-info"
                  >
                    <span>
                      4567
                    </span>
                    <div>
                      Total Documents
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </section>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-xs-12"
      >
        <section
          className="panel "
        >
          <header
            className="panel-heading"
          >
            Source
          </header>
          <div
            className="panel-body undefined"
            style={
              Object {
                "backgroundColor": "#FFF",
              }
            }
          />
        </section>
      </div>
    </div>
  </section>
</div>
`;
